inference-engine/thirdparty/clDNN/kernel_selector/core/actual_kernels/eltwise/eltwise_kernel_b_fs_yx_fsv16.cpp

```diff
@@ -52,31 +75,62 @@ JitConstants EltwiseKernel_b_fs_yx_fsv16::MakeLoadJitConstants(const eltwise_par
         for (size_t input_idx = 0; input_idx < ew.inputs.size(); input_idx++) {
             const auto &input = ew.inputs[input_idx];
             const std::string name = "INPUT_" + op_num_str + "_" + std::to_string(input_idx);
-            std::string idx_order = "INPUT" + std::to_string(input.index) + "_IDX_ORDER";
 
             switch (input.mode) {
                 case EltwiseInputMode::SCALAR:
                     jit.AddConstant(MakeJitConstant(name, input.scalar));
                     break;
                 case EltwiseInputMode::INPUT_BUFFER:
-                    if (params.inputs[input.index].LogicalSize() == params.output.Feature().v &&
-                        params.inputs[input.index].LogicalSize() == params.inputs[input.index].Feature().v) {
-                        jit.AddConstant(MakeJitConstant(name,
-                                                        "BLOCK_READN(INPUT" + std::to_string(input.index) + "_TYPE, 1, input" + std::to_string(input.index) +
-                                                        ", INPUT"+std::to_string(input.index)+"_GET_INDEX(b, f_block*16, y, x))"));
-                    } else if (params.inputs[input.index].LogicalSize() == 1) {
+                {
+                    const std::string idx_order = "INPUT" + std::to_string(input.index) + "_IDX_ORDER";
+                    jit.AddConstant(MakeJitConstant(idx_order, "b, f_block*16, y, x"));
+
+                    if (params.inputs[input.index].LogicalSize() == 1) {
                         jit.AddConstant(MakeJitConstant(name,
                                                         "input" + std::to_string(input.index) +
                                                         "[0]"));
                     } else {
-                        std::string block_read_str = "BLOCK_READN(INPUT" + std::to_string(input.index) + "_TYPE, " +
-                                                                 "BLOCK_SIZE, " +
-                                                                 "input" + std::to_string(input.index) + ", " +
-                                                                 "INPUT" + std::to_string(input.index) + "_GET_INDEX(b, f_block*16, y, x))";
-                        jit.AddConstant(MakeJitConstant(name,
-                                                        "TO_TYPE(MAKE_VECTOR_TYPE(ACCUMULATOR_TYPE, BLOCK_SIZE), " + block_read_str + ")"));
+                        bool feature_broadcasting = (params.inputs[input_idx].Feature().v == 1 && params.output.Feature().v != 1);
+
+                        if ((params.inputs[input.index].LogicalSize() == params.output.Feature().v &&
+                            params.inputs[input.index].LogicalSize() == params.inputs[input.index].Feature().v) ||
+                            (GetBlockSize(params) == 1)) {
```

Comment:
Why is it necessary to handle `GetBlockSize==1` case here? I think the statements below would work for this case as well.

Reply:
I'll update if/else branch as we talked it.